packages/framework/src/Model/Store/OpeningHours/StoreOpeningHoursProvider.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Store\OpeningHours;
+
+use DateTimeImmutable;
+use InvalidArgumentException;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Component\Localization\DisplayTimeZoneProviderInterface;
+use Shopsys\FrameworkBundle\Model\Store\ClosedDay\ClosedDayFacade;
+use Shopsys\FrameworkBundle\Model\Store\Store;
+use Spatie\OpeningHours\OpeningHours as SpatieOpeningHours;
+
+class StoreOpeningHoursProvider
+{
+    protected const DAY_NUMBERS_TO_ENGLISH_NAMES_MAP = [
+        1 => 'monday',
+        2 => 'tuesday',
+        3 => 'wednesday',
+        4 => 'thursday',
+        5 => 'friday',
+        6 => 'saturday',
+        7 => 'sunday',
+    ];
+
+    /**
+     * @var \Spatie\OpeningHours\OpeningHours[]
+     */
+    protected array $openingHoursSetting = [];
```

Comment:
I was thinking about that during the implementation and thought it was unnecessary here. On the other hand, now, with some time gap, I think it won't harm and maybe it will save someone from unexpected behavior in the future, so added in 5cee258c00